src/main/java/org/springframework/data/couchbase/repository/query/StringN1qlQueryCreator.java

```diff
@@ -153,9 +153,10 @@ private QueryCriteria from(final Part part, final CouchbasePersistentProperty pr
 	// copied from StringN1qlBasedQuery
 	private N1QLExpression getExpression(ParameterAccessor accessor, Object[] runtimeParameters,
 			ReturnedType returnedType) {
-		EvaluationContext evaluationContext = evaluationContextProvider.getEvaluationContext(
-				getQueryMethod().getParameters(), runtimeParameters);
-		N1QLExpression parsedStatement = x(this.queryParser.doParse(parser, evaluationContext, false));
+		boolean isCountQuery = queryMethod.getName().toLowerCase().startsWith("count"); // should be queryMethod.isCountQuery()
```

Comment:
is this still a todo?

Reply:
queryMethod.isCount() is in the datacouch-588 branch along with some others.